verification/src/compiler/version.rs

```diff
@@ -209,4 +266,141 @@ mod tests {
             )
         );
     }
+
+    fn to_versions(versions: Vec<&str>) -> Vec<CompilerVersion> {
+        versions
+            .iter()
+            .map(|s| CompilerVersion::from_str(s).expect("invalid version"))
+            .collect()
+    }
+
+    fn test_shuffle_and_sort(sorted: Vec<&str>) {
+        let sorted_versions = to_versions(sorted.clone());
+        let mut shuffled_versions = sorted_versions.clone();
+        shuffled_versions.shuffle(&mut thread_rng());
+        shuffled_versions.sort();
+        let shuffled: Vec<String> = shuffled_versions.iter().map(|v| v.to_string()).collect();
+        assert_eq!(shuffled, sorted);
+    }
```

Comment:
```suggestion
    fn test_shuffle_and_sort(sorted: Vec<&str>) {
        let to_versions: fn(Vec<&str>) -> Vec<CompilerVersion> = |versions| {
            versions
                .iter()
                .map(|s| CompilerVersion::from_str(s).expect("invalid version"))
                .collect()
        };
        let sorted_versions = to_versions(sorted.clone());
        let mut shuffled_versions = sorted_versions.clone();
        shuffled_versions.shuffle(&mut thread_rng());
        shuffled_versions.sort();
        let shuffled: Vec<String> = shuffled_versions.iter().map(|v| v.to_string()).collect();
        assert_eq!(shuffled, sorted);
    }
```

Nit: I believe putting `to_version` function inside a closure would be clearer to read, as `to_versions` is not used anywhere else